internal/mackerel/service.go

```diff
@@ -22,56 +22,53 @@ func ServiceNameValidator() validator.String {
 	)
 }
 
-type ServiceModel struct {
+type ServiceModel = ServiceModelV0
+
+type ServiceModelV0 struct {
 	ID   types.String `tfsdk:"id"`
-	Name types.String `tfsdk:"name"`
+	Name string       `tfsdk:"name"`
 	Memo types.String `tfsdk:"memo"`
 }
 
-func ReadService(ctx context.Context, client *Client, name string) (*ServiceModel, error) {
+func ImportService(_ context.Context, id string) (ServiceModel, error) {
+	return ServiceModelV0{
+		ID:   types.StringValue(id),
+		Name: id,
+	}, nil
+}
+
+func ReadService(ctx context.Context, client *Client, name string) (ServiceModel, error) {
 	return readServiceInner(ctx, client, name)
 }
 
 type serviceFinder interface {
 	FindServices() ([]*mackerel.Service, error)
 }
 
-func readServiceInner(_ context.Context, client serviceFinder, name string) (*ServiceModel, error) {
+func readServiceInner(_ context.Context, client serviceFinder, name string) (ServiceModel, error) {
 	services, err := client.FindServices()
 	if err != nil {
-		return nil, err
+		return ServiceModel{}, err
 	}
 
 	serviceIdx := slices.IndexFunc(services, func(s *mackerel.Service) bool {
 		return s.Name == name
 	})
 	if serviceIdx == -1 {
-		return nil, fmt.Errorf("the name '%s' does not match any service in mackerel.io", name)
+		return ServiceModel{}, fmt.Errorf("the name '%s' does not match any service in mackerel.io", name)
 	}
 
 	service := services[serviceIdx]
-	return &ServiceModel{
+	return ServiceModelV0{
 		ID:   types.StringValue(service.Name),
-		Name: types.StringValue(service.Name),
+		Name: service.Name,
 		Memo: types.StringValue(service.Memo),
 	}, nil
 }
 
-func (m *ServiceModel) Set(newData ServiceModel) {
-	if !newData.ID.IsUnknown() {
-		m.ID = newData.ID
-	}
-	m.Name = newData.Name
-
-	// If Memo is an empty string, treat it as null
-	if newData.Memo.ValueString() != "" {
-		m.Memo = newData.Memo
-	}
```

Comment:
`memo` is optional, but null values are NOT preserved and filled by empty string.